dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/header/HeaderExchangeHandler.java

```diff
@@ -209,10 +210,16 @@ public void received(Channel channel, Object message) throws RemotingException {
                     Exception e = new Exception("Dubbo client can not supported string message: " + message + " in channel: " + channel + ", url: " + channel.getUrl());
                     logger.error(e.getMessage(), e);
                 } else {
-                    String echo = handler.telnet(channel, (String) message);
-                    if (echo != null && echo.length() > 0) {
-                        channel.send(echo);
+                    if (TelnetHandlerAdapter.isEnableTelnet()){
+                        String echo = handler.telnet(channel, (String) message);
+                        if (echo != null && echo.length() > 0) {
+                            channel.send(echo);
+                        }
+                    }else{
+                        channel.send("telnet disabled");
+                        channel.close();

```

Comment:
Why we close channel here?

Reply:
I think since telnet configed to close, just close the telnet connected channel is fine when recived a telnet request

Reply:
There is a problem here. Even if it is a telnet client, it can still transfer the serialized request object. We can only control the server does not receive the string information, can not guess whether the client is telnet or dubbo client. Any data can be transferred via the tcp link.

Reply:
got it